cpp/src/parquet/column_reader.cc

```diff
@@ -337,6 +348,50 @@ void SerializedPageReader::UpdateDecryption(const std::shared_ptr<Decryptor>& de
   }
 }
 
+bool SerializedPageReader::ShouldSkipPage(EncodedStatistics& page_statistics) {
+  const PageType::type page_type = LoadEnumSafe(&current_page_header_.type);
+  if (page_type == PageType::DATA_PAGE) {
+    const format::DataPageHeader& header = current_page_header_.data_page_header;
+    CheckNumValuesInHeader(header.num_values);
+    page_statistics = ExtractStatsFromHeader(header);
```

Comment:
Sorry I can't find the comment, but do we need some way of filtering out known invalid statistics here, or is it already taken care of?

Reply:
ColumnChunkMetadata has is_stats_set(), which should be used to check if the statistics are correct, and if yes, set a callback function.